File: plotter.cpp
#include <vector>
#include "Point.h"
#include <fstream>
#include "plotter.h"

using std::vector;
void plot(vector<Point> path, vector<vector<Point>> polygons) {
    std::ofstream outFile("graph.txt");
    outFile << path.front().x << "_" <<  path.front().y << '\n';
    outFile << path.back().x << "_" << path.back().y << '\n';
    for (auto it = polygons.begin(); it != polygons.end(); it++) {
        for (auto jt = it->begin(); jt != it->end(); jt++) {
            outFile << jt->x << "_" << jt->y << " ";
        }
        outFile << '\n';
    }
    outFile << "PATH\n";

    for (auto it = path.begin(); it != path.end(); it++) {
        outFile << it->x << "_" << it->y << '\n';
    }
    outFile.close();
}

File: plotter.py
from matplotlib import pyplot as plt

f = open("graph.txt", "r")
startLine = f.readline().split('_')
startX = startLine[0]
startY = startLine[1]

goalLine = f.readline().split('_')
goalX = goalLine[0]
goalY = goalLine[1]

plt.plot(float(startX), float(startY), "g^", markersize=20)
plt.plot(float(goalX), float(goalY), "y*", markersize=20)

line = f.readline() 
while line.strip() != "PATH":
    line = line.split()
    x = []
    y = []
    for point in line:
        x.append(float(point.split('_')[0]))
        y.append(float(point.split('_')[1]))
    x.append(x[0])
    y.append(y[0])
    plt.plot(x, y, 'b')
    
    plt.xlim(0, 12)
    plt.ylim(0, 12)
    line = f.readline()

x = []
y = []
for line in f.readlines():
    line = line.split('_')
    x.append(float(line[0]))
    y.append(float(line[1]))

plt.plot(x, y, 'r--')

plt.show()
